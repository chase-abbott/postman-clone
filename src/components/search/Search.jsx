import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const SearchForm = styled.form`
display: flex;
flex-direction: column;
align-items: center;
width: 100%;`;

const SearchBar = styled.input`
width: 79.75%;
align-self: flex-end;
`;

const InputBar = styled.div`
display: flex;
width: 75%;
justify-content: space-evenly;
align-items: center;
`;

const RadioContainer = styled.div`
width: 75px;
height: 35px;
display: flex;
justify-content: center;
align-items: center;
`
;

const RadioLabel = styled.label`
`;

const RadioInput = styled.input`
cursor: pointer;

&:checked&label {
  background-color: #204060;
  color: white;
`;

const SubmitButton = styled.button`
height: 80%;
width: 50px;
`;

const JSONBody = styled.textarea`
width: 60%;
height: 100px;
`;

const Search = ({ 
  onMethodChange, 
  urlInput, 
  onTextChange, 
  onSubmit,
  body, 
  onBodyChange }) => {
  return (
    <SearchForm onSubmit={onSubmit}>
      <SearchBar
        data-testid="search-bar"
        type="text"
        name="urlInput"
        value={urlInput}
        onChange={onTextChange}>
      </SearchBar>
      <InputBar>
        <RadioContainer>
          <RadioLabel htmlfor="GET"> GET </RadioLabel>
          <RadioInput 
            type="radio" 
            value="GET" 
            name="selector" 
            onChange={onMethodChange}></RadioInput>
        </RadioContainer>
        <RadioContainer>
          <RadioLabel htmlfor="POST"> POST </RadioLabel>
          <RadioInput 
            type="radio" 
            value="POST" 
            name="selector" 
            onChange={onMethodChange}></RadioInput>
        </RadioContainer>
        <RadioContainer>
          <RadioLabel htmlfor="PUT"> PUT </RadioLabel>
          <RadioInput 
            type="radio" 
            value="PUT" 
            name="selector" 
            onChange={onMethodChange}></RadioInput>
        </RadioContainer>
        <RadioContainer>
          <RadioLabel htmlfor="PATCH"> PATCH </RadioLabel>
          <RadioInput 
            type="radio" 
            value="PATCH" 
            name="selector" 
            onChange={onMethodChange}></RadioInput>
        </RadioContainer>
        <RadioContainer>
          <RadioLabel htmlfor="DELETE"> DELETE </RadioLabel>
          <RadioInput 
            type="radio" 
            value="DELETE" 
            name="selector" 
            onChange={onMethodChange}></RadioInput>
        </RadioContainer>
        <SubmitButton> Go </SubmitButton>
      </InputBar>
      <JSONBody 
        placeholder="Raw JSON Body"
        name="body"
        value={body}
        onChange={onBodyChange}/>
    </SearchForm>
  );
};

Search.propTypes = {
  onMethodChange: PropTypes.func.isRequired,
  urlInput: PropTypes.string,
  onTextChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  body: PropTypes.string,
  onBodyChange: PropTypes.func.isRequired,
};

export default Search;
