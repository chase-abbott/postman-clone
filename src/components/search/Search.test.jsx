import React from 'react';
import { render } from '@testing-library/react';
import Search from './Search.jsx';

describe('Search component snapshot', () => {
  it('Renders a search component on the page', () => {
    const { asFragment } = render(<Search/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
