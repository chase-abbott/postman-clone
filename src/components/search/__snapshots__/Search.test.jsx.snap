// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component snapshot Renders a search component on the page 1`] = `
<DocumentFragment>
  <form
    class="sc-bdnxRM eNlDXM"
  >
    <input
      class="sc-gtsrHT kbxame"
      name="urlInput"
      type="text"
      value=""
    />
    <div
      class="sc-dlnjwi jlHniZ"
    >
      <div
        class="sc-hKFxyN eJFrPL"
      >
        <label
          class="sc-eCApnc btdohc"
        >
           GET 
        </label>
        <input
          class="sc-jSFjdj dAKPiI"
          name="selector"
          type="radio"
          value="GET"
        />
      </div>
      <div
        class="sc-hKFxyN eJFrPL"
      >
        <label
          class="sc-eCApnc btdohc"
        >
           POST 
        </label>
        <input
          class="sc-jSFjdj dAKPiI"
          name="selector"
          type="radio"
          value="POST"
        />
      </div>
      <div
        class="sc-hKFxyN eJFrPL"
      >
        <label
          class="sc-eCApnc btdohc"
        >
           PUT 
        </label>
        <input
          class="sc-jSFjdj dAKPiI"
          name="selector"
          type="radio"
          value="PUT"
        />
      </div>
      <div
        class="sc-hKFxyN eJFrPL"
      >
        <label
          class="sc-eCApnc btdohc"
        >
           PATCH 
        </label>
        <input
          class="sc-jSFjdj dAKPiI"
          name="selector"
          type="radio"
          value="PATCH"
        />
      </div>
      <div
        class="sc-hKFxyN eJFrPL"
      >
        <label
          class="sc-eCApnc btdohc"
        >
           DELETE 
        </label>
        <input
          class="sc-jSFjdj dAKPiI"
          name="selector"
          type="radio"
          value="DELETE"
        />
      </div>
      <button
        class="sc-gKAaRy dRChrM"
      >
         Go 
      </button>
    </div>
    <textarea
      class="sc-iCoGMd chwWKm"
      name="body"
      placeholder="Raw JSON Body"
    />
  </form>
</DocumentFragment>
`;
